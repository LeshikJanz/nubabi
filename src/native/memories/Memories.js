// @flow
import React from 'react';
import { Box } from '../components';
import AddMemoryHeader from './AddMemoryHeader';
import SuggestedMemories from './SuggestedMemories';
import ViewMemories from './ViewMemories';

type Props = {
  onAddMemory: () => void,
};

export const Memories = ({ onAddMemory }: Props) => {
  return (
    <Box flex={1}>
      <AddMemoryHeader onAddMemory={onAddMemory} />
      <ViewMemories />
    </Box>
  );
};

export default Memories;
