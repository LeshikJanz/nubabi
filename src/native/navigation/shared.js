import React from 'react';
import { pathOr } from 'ramda';
import LeftHeader from './LeftHeader';
import RightHeader from './RightHeader';

const babyNameTitle = ({ state }) => (
  pathOr('Nubabi', ['params', 'babyName'], state)
);

export function getBabyTitle() {
  return { title: babyNameTitle };
}

export const getTabHeaders = (navigate) => ({
  left: <LeftHeader navigate={navigate} />,
  right: <RightHeader navigate={navigate} />,
});
