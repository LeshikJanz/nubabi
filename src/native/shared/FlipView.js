import React, { Component } from 'react';
import {
  Animated,
  TouchableOpacity,
} from 'react-native';

const styles = {
  animatedContainer: {
    flex: 1,
  },
};

class FlipCard extends Component {
  constructor(props) {
    super(props);
    this._flipToggleCard = this._flipToggleCard.bind(this);
    this.state = {
      animatedValue: new Animated.Value(0),
      isFlipped: true,
    };
  }

  componentDidUpdate(prevProp, prevState) {
    if (this.state.isFlipped !== prevState.isFlipped) {
      this._flippedCard();
    }
  }

  _flipToggleCard() {
    this.setState({ isFlipped: !this.state.isFlipped });
  }

  _flippedCard() {
    Animated.spring(this.state.animatedValue, {
      toValue: 0,   // Returns to the start
      velocity: this.props.velocity,  // Velocity makes it move
      tension: this.props.tension, // Slow
      friction: this.props.friction,  // Oscillate a lot
    }).start();
  }

  flippedCardView(isFlipped) {
    if (isFlipped) {
      return this.props.renderFront;
    }
    return this.props.renderBack;
  }

  render() {
    const rotateX = this.state.animatedValue.interpolate({
      inputRange: [0, 1],
      outputRange: ['0deg', '180deg'],
    });
    return (
      <TouchableOpacity onPress={this._flipToggleCard} style={styles.animatedContainer}>
        <Animated.View
          style={[styles.animatedContainer, { transform: [{ rotateX }] }]}
        >
          {this.flippedCardView(this.state.isFlipped)}
        </Animated.View>
      </TouchableOpacity>);
  }
}

FlipCard.propTypes = {
  renderBack: React.PropTypes.object.isRequired,
  renderFront: React.PropTypes.object.isRequired,
  velocity: React.PropTypes.number.isRequired,
  tension: React.PropTypes.number.isRequired,
  friction: React.PropTypes.number.isRequired,
};

module.exports = FlipCard;
