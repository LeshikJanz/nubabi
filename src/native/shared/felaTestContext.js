import React from 'react';
import { NativeFelaProvider } from '../components/FelaProvider';
import { createRenderer } from 'fela-native';
import theme from '../../common/themes/defaultTheme';

export const renderer = createRenderer();
const FelaProvider = NativeFelaProvider(renderer, theme);

const felaTestContext = component =>
  <FelaProvider>
    {component}
  </FelaProvider>;

export default felaTestContext;
