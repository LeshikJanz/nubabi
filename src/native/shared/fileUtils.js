// @flow
import { getContentTypeFromFilename } from '../../common/helpers/graphqlUtils';

export const isNewFile = (file: ?{ url: string }) => {
  if (!file) {
    return false;
  }

  return !file.url.startsWith('https://firebasestorage.googleapis.com');
};

// Parse results from react-native-image-picker into File interface inout
// (do not confuse with react-native-image-crop-picker which already gives
// us the expected result
export const fileFromImagePickerResult = file => {
  return {
    size: file.fileSize,
    name: file.fileName,
    url: file.uri,
    contentType: getContentTypeFromFilename(file.fileName),
  };
};
