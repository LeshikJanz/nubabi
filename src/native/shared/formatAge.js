import moment from 'moment';
import pluralize from 'pluralize';

const pluralizeAge = (count, unit) => {
  return [count, pluralize(unit, count), 'old'].join(' ');
};

export const formatAge = dobString => {
  const now = moment();
  const dob = moment(dobString);

  const diff = now.diff(dob, 'months');
  const shouldDisplayWeeks = diff === 0;

  if (shouldDisplayWeeks) {
    const weeks = now.diff(dob, 'week');
    return pluralizeAge(weeks, 'week');
  }

  const shouldDisplayYears = diff >= 12;

  if (shouldDisplayYears) {
    const years = now.diff(dob, 'years');
    return pluralizeAge(years, 'year');
  }

  return pluralizeAge(diff, 'month');
};
