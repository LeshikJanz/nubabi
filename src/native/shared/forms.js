// @flow
import React from 'react';
import { View, Text, TextInput } from 'react-native';
import moment from 'moment';
import theme, { FONT_COLOR } from '../../common/themes/defaultTheme';
import RelationshipDropdown from '../profile/EditBaby/RelationshipDropdown';

export const required = (value: ?mixed) => {
  return value ? undefined : 'Required';
};

export const maxLength = (max: number) => (value: ?string) => {
  return value && value.length > max
    ? `Must be ${max} characters or less`
    : undefined;
};

export const minValue = (min: number) => (value: ?number) => {
  return value && value < min ? `Must be at least ${min}` : undefined;
};

export const maxValue = (max: number) => (value: ?number) => {
  return value && value > max ? `Must be at most ${max}` : undefined;
};

export const formattedDate = (format: string) => (value: ?Date) => {
  if (!value) {
    return;
  }

  if (!moment(value, format).isValid()) {
    // eslint-disable-next-line consistent-return
    return `Must be a date in ${format} format`;
  }
};

export const constantValues = (...constants: Array<string>) => (
  value: ?string,
) => {
  return value && !constants.includes(value)
    ? `Must be one the values ${constants.join(',')}`
    : undefined;
};

export const renderTextInput = field => {
  // we can access errors on field.meta.errors and dirty state and field.meta.touched
  const { label } = field;
  const { touched, error } = field.meta;

  const hasError = touched && error;
  // We don't want to show required as an error since the coloring
  // should suffice
  const hasExplicitError = hasError && error !== 'Required';

  const containerStyle = [
    styles.inputContainer,
    hasError ? styles.inputContainerError : {},
  ];

  const labelStyle = [styles.inputLabel, hasError ? styles.hasError : {}];

  return (
    <View style={containerStyle}>
      <View style={{ flexDirection: 'row', justifyContent: 'space-around' }}>
        {label
          ? <Text style={[...labelStyle, { flex: 1 }]}>
              {label}
            </Text>
          : null}
        {hasExplicitError
          ? <Text style={labelStyle}>
              {error.toUpperCase()}
            </Text>
          : null}
      </View>
      <TextInput {...field.input} style={styles.textInput} />
    </View>
  );
};

export const renderRelationshipDropdown = field => {
  return <RelationshipDropdown field={field} />;
};

const styles = {
  inputContainer: {
    flex: 1,
    flexDirection: 'column',
    height: 40,
    marginHorizontal: 30,
    marginBottom: 15,
    borderBottomWidth: 1,
    paddingBottom: 5,
    borderColor: '#eff1f7',
  },
  inputContainerError: {
    borderColor: theme.colors.danger,
  },
  hasError: {
    color: theme.colors.danger,
  },
  textInput: {
    flex: 1,
    color: FONT_COLOR,
    fontSize: 16,
  },
  inputLabel: {
    fontSize: 8,
    color: '#a8b3c2',
    marginBottom: 4,
  },
};
