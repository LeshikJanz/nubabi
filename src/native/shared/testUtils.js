// @flow
import testRenderer from 'react-test-renderer';
import felaTestContext, { renderer as felaRenderer } from './felaTestContext';

const felaNativeToStyleObject = cache => {
  return Object.keys(cache).reduce((acc, key) => {
    return { ...acc, ...JSON.parse(key) };
  }, {});
};

export const createTestRenderer = () => (Component: any) => {
  const component = testRenderer.create(felaTestContext(Component));

  const snapshot = component.toJSON();
  snapshot.felaRules = felaNativeToStyleObject(felaRenderer.cache);

  return snapshot;
};

export const createExpectRender = () => {
  const render = createTestRenderer();
  return (Component: any) => {
    expect(render(Component)).toMatchSnapshot();
  };
};

export const expectRender = createExpectRender();
