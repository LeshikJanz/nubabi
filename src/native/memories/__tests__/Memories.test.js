import 'react-native';
import React from 'react';
import { Memories } from '../Memories';
import { expectRender } from '../../shared/testUtils';

test('it renders correctly', () => {
  expectRender(<Memories />);
});
