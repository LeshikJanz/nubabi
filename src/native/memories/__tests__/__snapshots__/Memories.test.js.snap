// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
style = Object {
    "flex": 1,
    "borderColor": "#E9ECF4",
    "borderBottomWidth": 0,
    "borderLeftWidth": 0,
    "borderRightWidth": 0,
    "borderTopWidth": 0,
    "paddingBottom": 8,
    "paddingLeft": 16,
    "paddingRight": 16,
    "paddingTop": 8,
    "flexDirection": "row",
    "backgroundColor": "#FFFFFF",
    "alignItems": "center",
    "borderBottomLeftRadius": 4,
    "borderBottomRightRadius": 4,
    "borderTopLeftRadius": 4,
    "borderTopRightRadius": 4,
    "borderWidth": 1,
    "fontSize": 13,
    "lineHeight": 16,
    "color": "#454D56",
    "fontFamily": "System",
    "fontWeight": "600",
    "marginLeft": 8,
    "justifyContent": "center"
}

<View
  style={
    Object {
      "borderBottomWidth": 0,
      "borderColor": "#9EABBC",
      "borderLeftWidth": 0,
      "borderRightWidth": 0,
      "borderTopWidth": 0,
      "flex": 1,
    }
  }
>
  <View
    contentSpacing={true}
    style={
      Object {
        "backgroundColor": "#EDF1FA",
        "borderBottomWidth": 0,
        "borderColor": "#9EABBC",
        "borderLeftWidth": 0,
        "borderRightWidth": 0,
        "borderTopWidth": 0,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderBottomLeftRadius": 4,
          "borderBottomRightRadius": 4,
          "borderBottomWidth": 0,
          "borderColor": "#E9ECF4",
          "borderLeftWidth": 0,
          "borderRightWidth": 0,
          "borderTopLeftRadius": 4,
          "borderTopRightRadius": 4,
          "borderTopWidth": 0,
          "borderWidth": 1,
          "flex": 1,
          "flexDirection": "row",
          "opacity": 1,
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "justifyContent": "center",
            "marginRight": 5,
            "marginTop": 2,
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#EA3154",
                "fontSize": 24,
              },
              undefined,
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "borderBottomWidth": 0,
            "borderColor": "#9EABBC",
            "borderLeftWidth": 0,
            "borderRightWidth": 0,
            "borderTopWidth": 0,
            "color": "#454D56",
            "fontFamily": "System",
            "fontSize": 13,
            "fontWeight": "600",
            "lineHeight": 16,
          }
        }
      >
        ADD A MEMORY
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderBottomLeftRadius": 4,
          "borderBottomRightRadius": 4,
          "borderBottomWidth": 0,
          "borderColor": "#E9ECF4",
          "borderLeftWidth": 0,
          "borderRightWidth": 0,
          "borderTopLeftRadius": 4,
          "borderTopRightRadius": 4,
          "borderTopWidth": 0,
          "borderWidth": 1,
          "justifyContent": "center",
          "marginLeft": 8,
          "opacity": 1,
          "paddingLeft": 16,
          "paddingRight": 16,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#748294",
              "fontSize": 24,
            },
            undefined,
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;
